packages/amplify-graphql-api-construct-tests/src/__tests__/sql-models.test.ts

```diff
@@ -0,0 +1,194 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import {
+  createNewProjectDir,
+  deleteDBInstance,
+  deleteDbConnectionConfig,
+  deleteProjectDir,
+  extractVpcConfigFromDbInstance,
+  setupRDSInstanceAndData,
+  storeDbConnectionConfig,
+} from 'amplify-category-api-e2e-core';
+import generator from 'generate-password';
+import { initCDKProject, cdkDeploy, cdkDestroy } from '../commands';
+import { graphql } from '../graphql-request';
+
+jest.setTimeout(1000 * 60 * 60 /* 1 hour */);
+
+interface DBDetails {
+  endpoint: string;
+  port: number;
+  dbName: string;
+  vpcConfig: {
+    vpcId: string;
+    securityGroupIds: string[];
+    subnetAvailabilityZones: {
+      subnetId: string;
+      availabilityZone: string;
+    }[];
+  };
+  ssmPaths: {
+    hostnameSsmPath: string;
+    portSsmPath: string;
+    usernameSsmPath: string;
+    passwordSsmPath: string;
+    databaseNameSsmPath: string;
+  };
+}
+
+describe('CDK GraphQL Transformer', () => {
+  let projRoot: string;
+  const projFolderName = 'sqlmodels';
+
+  const [username, password, identifier] = generator.generateMultiple(3);
+
+  const region = process.env.AWS_REGION ?? 'us-east-1';
+
+  const dbname = 'default_db';
+
+  let dbDetails: DBDetails;
+
+  beforeAll(async () => {
+    dbDetails = await setupDatabase({
+      identifier,
+      engine: 'mysql',
+      dbname,
+      username,
+      password,
+      region,
+    });
+  });
+
+  afterAll(async () => {
+    await cleanupDatabase({ identifier: identifier, region, dbDetails });
+  });
+
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir(projFolderName);
+  });
+
+  afterEach(async () => {
+    try {
+      const result = await cdkDestroy(projRoot, '--all');
+      console.log(`cdkDestroy result: ${JSON.stringify(result)}`);
+    } catch (_) {
+      /* No-op */
+    }
+
+    deleteProjectDir(projRoot);
+  });
+
+  it('creates a GraphQL API from SQL-based models', async () => {
+    const templatePath = path.resolve(path.join(__dirname, 'backends', 'sql-models'));
+    const name = await initCDKProject(projRoot, templatePath, 'latest');
+    writeDbDetails(dbDetails, projRoot);
```

Comment:
Was hoping there would be a pass these to the App via context or app environment, but my brief web search did not point me to anything useful.